src/gretel_trainer/relational/json.py

```diff
@@ -129,6 +127,34 @@ def make_suffix(s):
     return hashlib.sha256(s.encode("utf-8")).hexdigest()[:10]
 
 
+def jsonencode(df: pd.DataFrame, cols: list[str]) -> pd.DataFrame:
+    def _jsonencode(x):
+        if isinstance(x, str):
+            return x
+        elif isinstance(x, (dict, list)):
+            return dumps(x)
+
+    copy = df.copy()
+    for col in cols:
+        copy[col] = copy[col].map(_jsonencode)
```

Comment:
Would it make sense to add a check for empty `cols` and returning the original df?

My thinking is that JSON cols won't be present in many datasets, and creating that is not needed in such cases (and in the context of memory consumption, maybe this could make some difference for large datasets?).

It does change the contract, as the function would return a copy in some cases and the original in others, so that's not great, but maybe worth considering, since this is quite internal detail and could be documented? Based on the current usage, it looks like this wouldn't cause any problems.

Reply:
Nice idea, I've added it